src/verify.js

```diff
@@ -0,0 +1,23 @@
+// @ts-check
+const fs = require("fs");
+const util = require('util');
+
+const fsStat = util.promisify(fs.stat);
+const fsReadFile = util.promisify(fs.readFile);
+
+/**
+ * @param {string} filename
+ * @param {string} content
+ * @returns {Promise}
+ */
+module.exports = async (filename, content) => {
+  return fsStat(filename).catch(() => {
+    throw new Error(`Verification has failed: ${filename} not found`);
+  }).then(() => {
+    return fsReadFile(filename, "utf-8").then((currentContent) => {
+      if (currentContent !== content) {
+        throw new Error(`Verification has failed: ${filename} content has changed`);
+      }
+    })
+  });
+};
```

Comment:
Since it's an async method, the code can rewritten using async/await instead of promises. I'd also be keen to update the error message to include an instruction on how to fix it so it's clear how to fix the issue for a person looking at the build failure without internal knowledge of this loader. What do you think @vhiairrassary ?
```suggestion
 // @ts-check
const fs = require('fs');
const util = require('util');

const fsExists = util.promisify(fs.exists);
const fsReadFile = util.promisify(fs.readFile);

/**
 * @param {string} filename
 * @param {string} content
 * @returns {Promise<void>}
 */
module.exports = async (filename, content) => {
    const fileExists = await fsExists(filename);
    if (!fileExists) {
        throw new Error(`Verification failed: Generated typings for css-module file '${filename}' is not found. It typically happens when the generated typings are not included in a commit.`);
    }

    const existingFileContent = await fsReadFile(filename, 'utf-8');
    // let's not fail the build if there are whitespace changes only
    if (existingFileContent.replace(/\s+/g, "") !== content.replace(/\s+/g, "")) {
        throw new Error(`Verification failed: Generated typings for css-modules file '${filename}' is out of date. It typically happens when the generated typings are not included in a commit.`);
    }
};
```

Reply:
I applied the suggestions (async/await & error messages) but I reverted to `fs.stat` instead of `fs.exists` since [it is deprecated](https://nodejs.org/api/fs.html#fs_fs_exists_path_callback).